### DISCUSSION TITLE
[RFC] Add paddingXXXS token

### DISCUSSION THREAD

**Discussion by guoyunhe** [ant-design/ant-design#51643] [RFCs]
Current smallest size token paddingXXS is 4px. However, antd does use 2px in some places. To get 2px, we get this kind of code:

```js
// components/date-picker/style/token.ts

// FIXED_ITEM_MARGIN is a hardcode calculation since calc not support rounding
const INTERNAL_FIXED_ITEM_MARGIN = Math.floor(paddingXXS / 2);
```

So how about create a token paddingXXXS that is 2px by default?

**Comment by petercat-assistant** [comment]
It seems like you're suggesting the addition of a new token, `paddingXXXS`, with a default value of 2px to the Ant Design system. This would help standardize the use of 2px padding across the library, rather than relying on calculations like `Math.floor(paddingXXS / 2)`.

To implement this, you would need to:

1. Define the new token `paddingXXXS` in the relevant style or theme configuration files.
2. Replace instances where 2px padding is used or calculated with the new `paddingXXXS` token.

This change would require updating the Ant Design codebase. If you are contributing to the project, you can create a pull request with these changes. If you are suggesting this feature to the Ant Design team, you might want to open a discussion or issue on their GitHub repository to propose this enhancement.

If you need further assistance or guidance on how to proceed with this, feel free to ask!

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by zombieJ** [comment]
cc @arvinxx

**Comment by arvinxx** [comment]
Why need 2px

**Reply by guoyunhe** [reply]
Because it is used in the code base, to separate some small elements, like tags in a Select.

**Reply by arvinxx** [reply]
token is used for autosacle based on initial value. Do you think your use case need to auto change with initial value change? 

for example. as the base size unit or site step change, should this value change with it?

**Reply by guoyunhe** [reply]
As in the source code, it does change as the token change. To be clear, it is NOT my use case. It is from the source code of ant-design repo...

```
// components/date-picker/style/token.ts

// FIXED_ITEM_MARGIN is a hardcode calculation since calc not support rounding
const INTERNAL_FIXED_ITEM_MARGIN = Math.floor(paddingXXS / 2);
```
